src/app/execution/txLogger.js

```diff
@@ -178,7 +178,7 @@ class TxLogger {
     })
 
     this._deps.txListener.event.register('newBlock', (block) => {
-      if (!block.transactions.length) {
+      if (!block.transactions || block.transactions && !block.transactions.length) {
```

Comment:
Could you add parenthesis for better readibility : `if (!block.transactions || (block.transactions && !block.transactions.length))`

Reply:
are you sure? `console.log(!![])` return true

Reply:
Sorry, you're right...
To have only one boolean operation you would have to do : 
```javascript
if (block.transactions && block.transactions.length) return
this.logEmptyBlock({ block })
```